### ISSUE TITLE
Unable to launch autokey without sudo

### ISSUE THREAD

**Issue by johncarterofmars** [autokey/autokey#931]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

Crash/Hang/Data loss

### Choose one or more terms that describe this issue:

- [ ] autokey triggers
- [X] autokey-gtk
- [ ] autokey-qt
- [ ] beta
- [ ] bug
- [ ] critical
- [ ] development
- [ ] documentation
- [ ] enhancement
- [ ] installation/configuration
- [ ] phrase expansion
- [ ] scripting
- [ ] technical debt
- [ ] user interface

### Which Linux distribution did you use?

Mint 20.3

### Which AutoKey GUI did you use?

GTK

### Which AutoKey version did you use?

I have tried both GTK and QT

### How did you install AutoKey?

I have tried directly from Ubuntu repositories and also downloading the deb packages from here - github

### Can you briefly describe the issue?

I have used Autokey for years without issue. Today, I had some system updates and then later, noticed that Autokey wasn't running. I ran it mannually and received these errors:
`
Traceback (most recent call last):
  File "/usr/bin/autokey-gtk", line 6, in <module>
    from pkg_resources import load_entry_point
ImportError: cannot import name 'load_entry_point' from 'pkg_resources' (unknown location)
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 153, in apport_excepthook
    with os.fdopen(os.open(pr_filename,
FileNotFoundError: [Errno 2] No such file or directory: '/var/crash/_usr_bin_autokey-gtk.1000.crash'

Original exception was:
Traceback (most recent call last):
  File "/usr/bin/autokey-gtk", line 6, in <module>
    from pkg_resources import load_entry_point
ImportError: cannot import name 'load_entry_point' from 'pkg_resources' (unknown location)
`
I uninstalled and reintalled several times, trying both qt and gtk versions with the same results. 

However, if I run with sudo, it will run, but of course as the root user. It seems that today's updates included updates to python3 but this has never been an issue before. 
I am guessing I have something configured incorrectly with Python3 that has caused this. 
Any advice? 

### Can the issue be reproduced?

Always

### What are the steps to reproduce the issue?

Install, run autokey

### What should have happened?

Ran successfully

### What actually happened?

error - 

`
Traceback (most recent call last):
  File "/usr/bin/autokey-gtk", line 6, in <module>
    from pkg_resources import load_entry_point
ImportError: cannot import name 'load_entry_point' from 'pkg_resources' (unknown location)
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 153, in apport_excepthook
    with os.fdopen(os.open(pr_filename,
FileNotFoundError: [Errno 2] No such file or directory: '/var/crash/_usr_bin_autokey-gtk.1000.crash'

Original exception was:
Traceback (most recent call last):
  File "/usr/bin/autokey-gtk", line 6, in <module>
    from pkg_resources import load_entry_point
ImportError: cannot import name 'load_entry_point' from 'pkg_resources' (unknown location)
`

### Do you have screenshots?

No

### Can you provide the output of the AutoKey command?

```bash
included above
```

### Anything else?

nothing

**Comment by josephj11** [comment]
Please note that we don't certify AutoKey for off world use. :)

This is definitely something new. I will look into it a bit later when I have the time.

I'm not familiar with Mint versions. I'm wondering if maybe you are using Python 3.12 which is something we haven't tried yet.

Running AutoKey as root is not a good idea. If doing so created or modified any files, they will now be owned as root and your normal user won't have full permissions to access them. You need to figure out if that happened and fix those files with chown, etc.

What happens if you run AutoKey using the [zero install method](https://github.com/autokey/autokey/wiki/Installing#zero-installation-method) and explicitly use an earlier version of Python?

Start with step 5 because the rest is already done and use python3.10 or python3.11 instead of python3 in the command. (You can run `ls /usr/bin/py*` to see what explicit versions of Python are setup on your system.)

We have almost no developers at the moment, so the experts may not be available to help us figure this out.

Notes:

pkg_resources is part of Python, not part of AutoKey.

This appears to be related to things like setuptools, pip, and Pypi which don't seem involved here.

**Comment by johncarterofmars** [comment]
I appreciate it. It's very helpful up here. Bet you never thought your creation would make it this far!

**Comment by josephj11** [comment]
Edited my previous post, so please reread it.

**Comment by johncarterofmars** [comment]
Thank you. The zero install didn't work initially, so I reinstalled all of the dependencies. That seemed to make it work. However, the installed version still gives the same errors. So I decided to uninstall and try again. Same result. 
I, then, restored a backup from Sunday - a time when I know Autokey worked, and it still didn't work after restoring. I used Timeshift which doesn't touch the home dir so that leads me to believe there is something broken in the home dir. 
Regardless, I can make it run from the terminal. It keeps the terminal open now. Is there a way to get the prompt back? I tried & at the end, but that didn't work. 
I appreciate it.

**Comment by johncarterofmars** [comment]
Is there a manual way to install the msi and then have a config file to tell it where to connect to?

**Comment by josephj11** [comment]
Did you try explicitly running an older version of Python? What is your default version of Python?

I just started my python interpreter and found out that I'm running 3.12. Didn't realize that upgraded default had happened. My AutoKey works fine, so that's probably not the problem.

Eliminating the obvious: when you run a background task (&), sometimes the prompt doesn't reappear until you press Enter again. Did you do that? What happens if you type Ctrl+Z - which should send the current task to the background? Does AutoKey terminate if you close the terminal?

I wonder if running `ps -ef | grep 'autokey'` would tell you anything useful about which process AutoKey is and which process started it. I'm not sure how to leverage this, but it's where I would start looking.

If AutoKey ran to start with, it's quite strange that you had to reinstall any dependencies. Only the optional ones might not already be there and they don't stop it from running.

**Comment by johncarterofmars** [comment]
python3 --version
3.8.10

And now I feel silly. I did do CTRL+Z and it kept running. So I killed it and then reran 
`
python3 -m autokey.gtkui -c &
`
hit enter, and sure enough, got my prompt back. Thank you. 
`
ps -ef | grep 'autokey'
user    136225    3639  1 16:21 pts/0    00:00:02 python3 -m autokey.gtkui -c
user    136678  136664  0 16:24 pts/6    00:00:00 grep --color=auto autokey
`
I think I have been using Autokey for about 3 years now, up here, without a single hiccup. All the Jeddaks here love it. This is the first time and its strange how it happened suddenly. I'll spool up a VM of Mint over the holidays and test it out. I really don't think its Autkey, something changed on my system but I just can't track it down yet. 

I really appreciate your help on this.
